include/seastar/core/abort_source.hh

```diff
@@ -65,8 +66,14 @@ public:
             as._subscriptions->push_back(*this);
         }
 
-        void on_abort() noexcept {
-            _target();
+        struct naive_cb_tag {}; // to disambiguate constructors
```

Comment:
The `.subscribe()` is now a template. Won't it "select" the correct constructor itself?

Reply:
Unfortunately clang isn't smart enough to disambuate the two constructors without this tag. I think it's because they both accept a templated noncopyable function.

Reply:
Add concept in v2 to make it clearer to the user